web/src/enterprise/campaigns/create/CreateCampaignPage.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useMemo } from 'react'
+import { PageTitle } from '../../../components/PageTitle'
+import { CampaignHeader } from '../detail/CampaignHeader'
+import MagnifyIcon from 'mdi-react/MagnifyIcon'
+
+const campaignSpec = `name: hello-world
+description: Add Hello World to READMEs
+
+# Find all repositories that contain a README.md file.
+on:
+  - repositoriesMatchingQuery: file:README.md
+
+# In each repository, run this command. Each repository's resulting diff is captured.
+steps:
+  - run: echo Hello World | tee -a $(find -name README.md)
+    container: alpine:3
+
+# Describe the changeset (e.g., GitHub pull request) you want for each repository.
+changesetTemplate:
+  title: Hello World
+  body: My first campaign!
+  branch: hello-world # Push the commit to this branch.
+  commit:
+    message: Append Hello World to all README.md files
+  published: false`
+
+const sourceApplyCommand = 'src campaign apply -f hello-world.campaign.yaml -preview'
+
+export interface CreateCampaignPageProps {
+    // Nothing for now, but using it so once this changes we get type errors in the routing files.
+}
+
+export const CreateCampaignPage: React.FunctionComponent<CreateCampaignPageProps> = () => {
+    const helloWorldDownloadUrl = useMemo(() => 'data:text/plain;charset=utf-8,' + encodeURIComponent(campaignSpec), [])
```

Comment:
I'm curious... can't you just compute this?
```suggestion
    const helloWorldDownloadUrl = 'data:text/plain;charset=utf-8,' + encodeURIComponent(campaignSpec)
```

And pull it outside of the component?

Reply:
Could, but then it would run at startup, not only when the component is actually used. That's pretty minor, but when we get to a point in the future where the webapp grows really big, it can become troublesome.

Reply:
Keep in mind different areas of the app are lazy-loaded, so it would only run on the startup of the campaigns area, not when only navigating somewhere else

Reply:
I don't think that's a real concern, but even if, you could still get rid of the `useMemo` right?

Reply:
> Keep in mind different areas of the app are lazy-loaded, so it would only run on the startup of the campaigns area, not when only navigating somewhere else

still unnecessary to keep in memory, no? 😬 I'll still change it ..

Reply:
Yeah, just wanted to point out that the at-startup argument is not accurate. I don't have a stake here and you have the most context @eseliger so you should own the decision